juju/juju
# introduce cached models for api server

This work hooks up the previously landed modelcache worker (#9580) and the core cache (#9556).

Adds the modelcache to the machine agent's manifold setup and is a dependency for the apiserver.

To illustrate the new model config watcher, the logging config facade is updated to use both get the model config from the cache, and uses the new config watcher to only watch the 'logging-config' value from the model configuration.

## QA
- bootstrap a new lxd controller
- deploy an ubuntu-lite unit in the default model
- update the model-config a few times to update the logging-config and observe in the logs that they are changed.
